httpcache.go

```diff
@@ -184,6 +184,12 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 			cachedResp.StatusCode = http.StatusOK
 
 			resp = cachedResp
+		} else if err != nil && req.Method == "GET" && canStaleOnError(cachedResp.Header, req.Header) {
+			// In case of transport failure and stale-if-error activated, returns cached content
+			// when available
+			cachedResp.Status = fmt.Sprintf("%d %s", http.StatusOK, http.StatusText(http.StatusOK))
+			cachedResp.StatusCode = http.StatusOK
```

Comment:
@rs can you set a new custom header so the caller can tell if this came from a stale cache. Remember to not use `X-` as it's deprecated. Ref: [RFC6648](https://tools.ietf.org/html/rfc6648)